### PR TITLE
HP-1699 add user agent header

### PR DESCRIPTION
**Pull request by george42-ctds**
JIRA ticket: [HP-1699](https://ctds-planx.atlassian.net/browse/HP-1699)

### Improvements

* add User-Agent header to fetch_token and revoke_token

### Deployment changes

[HP-1699]: https://ctds-planx.atlassian.net/browse/HP-1699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mfshao — COMMENTED** [review]
looks cool 👍 just some minor comments

File: wts/blueprints/oauth2.py [hunk 1]
```diff
@@ -95,7 +95,10 @@ def logout_oauth():
     client = get_oauth_client(idp="default")
 
     try:
-        client.session.revoke_token(url, token)
+        app_version = flask.current_app.config.get("APP_VERSION", "0.0.0")
+        client.session.revoke_token(
+            url, token, headers={"User-Agent": f"Gen3WTS / {app_version}"}
```
[the remark this answers is not in the document]
Comment (george42-ctds) [reply wts/blueprints/oauth2.py]: Shoot, that isn't working for me. I'm getting an "Invalid username" message from https://idp.stage.qdr.org/ when I try to log in.

Can you verify if you are able to log in or not? The change is deployed in QA-HEAL.

I had tried removing the spaces in [this earlier commit](https://github.com/uc-cdis/workspace-token-service/pull/92/commits/99469dc9f8687d6c866db9eaf129a761735ebaff). I wasn't able to log in so I had reverted the change in the next commit.

[the remark this answers is not in the document]
Comment (mfshao) [reply wts/blueprints/oauth2.py]: Just tried in QA, I could refresh my QDR creds after logging in via their staging env
Have you regierstered an account at https://stage.qdr.org/ ?

[the remark this answers is not in the document]
Comment (george42-ctds) [reply wts/blueprints/oauth2.py]: Yes, it is now working for me! I'm able to log in to stage after being redirected from the qa-heal authorize_url endpoint. I'm not sure why it wasn't working earlier. 

I will update the unit tests to remove the spaces in the expected header value.

**Review by mfshao — APPROVED** [review]
👍

**Pull request merged**